app/uk/gov/hmrc/fileupload/frontendGlobal.scala

```diff
@@ -16,94 +16,116 @@
-    Logger.info(s"Starting frontend : $appName : in mode : ${app.mode}")
-    super.onStart(app)
-    ApplicationCrypto.verifyConfiguration()
+  lazy val healthRoutes = new manualdihealth.Routes(httpErrorHandler, new uk.gov.hmrc.play.health.AdminController(configuration))
+  lazy val appRoutes = new app.Routes(httpErrorHandler, fileUploadController)
 
-    // event stream
-    import play.api.Play.current
-    import play.api.libs.concurrent.Akka
-    val eventStream = Akka.system.eventStream
-    subscribe = eventStream.subscribe
-    publish = eventStream.publish
+  lazy val adminRoutes = new admin.Routes(httpErrorHandler, adminController)
 
-    notifyAndPublish = NotifierService.notify(sendNotification, publish) _
+  lazy val metrics = new GraphiteMetricsImpl(applicationLifecycle, configuration)
+  lazy val metricsController = new MetricsController(metrics)
 
-    // scanner
-    Akka.system.actorOf(ScannerActor.props(subscribe, scanBinaryData, notifyAndPublish), "scannerActor")
-    Akka.system.actorOf(TransferActor.props(subscribe, streamTransferCall), "transferActor")
+  lazy val prodRoutes = new prod.Routes(httpErrorHandler, new Provider[MetricsController] {
+    override def get(): MetricsController = metricsController
+  }, healthRoutes, appRoutes, adminRoutes)
 
-    fileUploadController
-    adminController
-    testOnlyController
-  }
+  override def router = if (configuration.getString("application.router").get == "testOnlyDoNotUseInAppConf.Routes") testRoutes else prodRoutes
 
+  lazy val fileUploadController = new FileUploadController(withValidEnvelope = withValidEnvelope, uploadParser = uploadParser,
+    notify = notifyAndPublish, now = now)
 
-  override def onStop(app: Application): Unit = {
-    super.onStop(app)
-  }
+  lazy val fileUploadBackendBaseUrl = baseUrl("file-upload-backend")
 
-  override def onLoadConfig(config: Configuration, path: java.io.File, classloader: ClassLoader, mode: Mode): Configuration = {
-    super.onLoadConfig(config, path, classloader, mode)
-  }
+  lazy val healthController = new uk.gov.hmrc.play.health.AdminController(configuration)
 
-  override def microserviceMetricsConfig(implicit app: Application): Option[Configuration] = app.configuration.getConfig(s"microservice.metrics")
+  lazy val testOnlyController = new TestOnlyController(fileUploadBackendBaseUrl, recreateCollections, wsClient)
 
-  // db
+  lazy val testRoutes = new testOnlyDoNotUseInAppConf.Routes(httpErrorHandler, testOnlyController, prodRoutes)
+
+  lazy val adminController = new AdminController(getFileInfo = getFileInfo, getChunks = getFileChunksInfo)(notify = notifyAndPublish)
 
-  lazy val db = DefaultMongoConnection.db
+  var subscribe: (ActorRef, Class[_]) => Boolean = _
+  var publish: (AnyRef) => Unit = _
+  var notifyAndPublish: (AnyRef) => Future[NotifyResult] = _
+  val now: () => Long = () => System.currentTimeMillis()
+
+  //TODO uncomment below
+  //ApplicationCrypto.verifyConfiguration()
```

Comment:
I think so as we don't use any encrypted key in our `application.conf` which is what [this library](https://github.com/hmrc/crypto) is used for.